log_test.go

```diff
@@ -78,6 +81,57 @@ func TestInfo(t *testing.T) {
 	})
 }
 
+func TestFatal(t *testing.T) {
+	t.Run("should exit", func(t *testing.T) {
+		if os.Getenv("TEST_FATAL") == "1" {
+			log := New(os.Stderr)
+			log.Fatal().Msg("")
+			return
+		}
+
+		cmd := exec.Command(os.Args[0], "-test.run=TestFatal/should_exit")
+		cmd.Env = append(os.Environ(), "TEST_FATAL=1")
+		stderr, err := cmd.StderrPipe()
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = cmd.Start()
+		if err != nil {
+			t.Fatal(err)
+		}
+		out, err := io.ReadAll(stderr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = cmd.Wait()
+		if err == nil {
+			t.Error("Expected log Fatal to exit with non-zero status")
+		}
```

Comment:
You should take a look at

- https://pkg.go.dev/os/exec#Cmd.Wait

- https://pkg.go.dev/os/exec#ExitError

- https://pkg.go.dev/os#ProcessState.ExitCode


This way you could to something like this


```suggestion
		err = cmd.Wait()
		if err == nil {
			t.Error("Expected test to fail")
		}

		var errExit exec.ExitError
		if !errors.As(err, &errExit) {
			t.Error("Expected an os.ExitError")
		}

		if errExit.ExitCode() != 1 {
			t.Error("Expected log Fatal to exit with non-zero status")
		}
```